src/components/Utils/filters.js

```diff
@@ -31,3 +31,90 @@ export const mergeFilters = (userFilter, adminAppliedPreFilter) => {
 
   return filterAB;
 };
+
+/**
+   * This function updates the counts in the initial set of tab options
+   * calculated from unfiltered data.
+   * It is used to retain field options in the rendering even if
+   * those options have no histogram results.
+   */
+export const updateCountsInInitialTabsOptions = (initialTabsOptions, processedTabsOptions) => {
+  const updatedTabsOptions = JSON.parse(JSON.stringify(initialTabsOptions));
+  const initialFields = Object.keys(initialTabsOptions);
+  for (let i = 0; i < initialFields.length; i += 1) {
+    const fieldName = initialFields[i];
+    const initialFieldOptions = initialTabsOptions[fieldName].histogram.map(x => x.key);
+    let processedFieldOptions = [];
+    if (Object.prototype.hasOwnProperty.call(processedTabsOptions, fieldName)) {
+      processedFieldOptions = processedTabsOptions[fieldName].histogram.map(x => x.key);
+    }
+
+    for (let j = 0; j < initialFieldOptions.length; j += 1) {
+      const optionName = initialFieldOptions[j];
+      let newCount;
+      if (processedFieldOptions.includes(optionName)) {
+        newCount = processedTabsOptions[fieldName].histogram.filter(
+          x => x.key === optionName,
+        )[0].count;
+      } else {
+        newCount = 0;
+      }
+      for (let k = 0; k < updatedTabsOptions[fieldName].histogram.length; k += 1) {
+        const option = updatedTabsOptions[fieldName].histogram[k];
+        if (option.key === optionName) {
+          updatedTabsOptions[fieldName].histogram[k].count = newCount;
+        }
+      }
+    }
+  }
+  return updatedTabsOptions;
+};
+
+function sortNumber(a, b) {
+  return a - b;
+}
+
+export const sortTabsOptions = (tabsOptions) => {
+  const fields = Object.keys(tabsOptions);
+  const sortedTabsOptions = {};
+  for (let x = 0; x < fields.length; x += 1) {
+    let field = fields[x];
+
+    const optionsForThisField = tabsOptions[field].histogram;
+
+    // Make { count -> [keys with this count] } dictionary
+    const countToKeys = {};
+    for (let i = 0; i < optionsForThisField.length; i += 1) {
+      let keyName = optionsForThisField[i].key;
+      let count = optionsForThisField[i].count;
+      if (countToKeys.hasOwnProperty(count)) {
+        countToKeys[count].push(keyName);
+      } else {
+        countToKeys[count] = [ keyName ];
+      }
+    }
+
+    // Sort the keys in each count
+    const countKeys = Object.keys(countToKeys);
+    countKeys = countKeys.map(x => parseInt(x));
+    for (let j = 0; j < countKeys.length; j += 1) {
+      countToKeys[countKeys[j]].sort(); // Alphabetically ascending order
+    }
+
+    // Sort the count-groups
+    countKeys.sort(sortNumber);
+    countKeys.reverse(); // Numerically descending order
```

Comment:
why don't just change the logic in `sortNumber`, then you don't need to use `reverse` here